common/threadable.h

```diff
@@ -28,32 +29,39 @@ class Threadable
     //! Run on the current thread, blocking until process ends
     virtual void run() = 0;
 
+    //! Check if the run() function has been called
+    virtual bool isRunning()
+    {
+        return m_DoRun;
+    }
+
     //! Run the process on a background thread
     virtual void runInBackground()
     {
-        m_Thread = std::thread([this] { run(); });
-        m_ThreadRunning = true;
+        m_Thread = std::make_unique<std::thread>([this] { run(); });
     }
 
     //! Stop the background thread
     virtual void stop()
     {
         m_DoRun = false;
-        if (m_ThreadRunning) {
-            if (m_Thread.joinable()) {
-                m_Thread.join();
+        if (m_Thread) {
+            if (m_Thread->joinable()) {
```

Comment:
``m_Thread->joinable()`` "Checks if the thread object identifies an active thread of execution." - if it doesn't shouldn't you just leave it alone rather than detach it? (which presumably only makes sense if it's also an active thread of execution - in which case you probably don't want to detach it anyway)

Reply:
I *believe* this is what you're supposed to do with threads -- either join them or detach them when you're finished with them. Maybe it's so you can free the resources associated with it? Not sure: https://en.cppreference.com/w/cpp/thread/thread/detach

It's what my C++ book suggests in any case and I was just blindly following it ;-)

Reply:
It looks like you're right. I double-checked in the book and it turns out I was misremembering it :-/